### ISSUE TITLE
SM arches 5.3, 6.2, 7.2 fail to build with RDC enabled.

### ISSUE THREAD

**Issue by allisonvacanti** [thrust/thrust#1167]
There are linker errors when any of those three architectures are targeted with RDC enabled. I've seen this on linux and windows, and also with nvcc 10.2 and 11.0.

```
nvcc -forward-unknown-to-host-compiler -ccbin=/usr/bin/c++ -gencode arch=compute_30,code=sm_30 -gencode arch=compute_32,code=sm_32 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_53,code=sm_53 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_62,code=sm_62 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_72,code=sm_72 -gencode arch=compute_75,code=sm_75 -gencode arch=compute_75,code=compute_75 -Xcompiler=-Werror -Xcompiler=-Wall -Xcompiler=-Wextra -Xcompiler=-Winit-self -Xcompiler=-Woverloaded-virtual -Xcompiler=-Wcast-qual -Xcompiler=-Wno-cast-align -Xcompiler=-Wno-long-long -Xcompiler=-Wno-variadic-macros -Xcompiler=-Wno-unused-function -Xcompiler=-Wno-unused-variable -Xcompiler=-Wlogical-op -Xcompiler=-Wno-noexcept-type -Werror all-warnings -Xcudafe --display_error_number -O3 -DNDEBUG -Xcompiler=-fPIC -Wno-deprecated-gpu-targets -shared -dlink testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o -o CMakeFiles/thrust.test.cuda.fill.dir/cmake_device_link.o  testing/libthrust.test.framework.a  -lcudadevrt  -lcudart_static
nvlink error   : Undefined reference to 'cudaGetParameterBuffer' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
nvlink error   : Undefined reference to 'cudaLaunchDevice' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
nvlink error   : Undefined reference to 'cudaGetLastError' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
nvlink error   : Undefined reference to 'cudaGetDevice' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
nvlink error   : Undefined reference to 'cudaGetErrorName' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
nvlink error   : Undefined reference to 'cudaGetErrorString' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
nvlink error   : Undefined reference to 'cudaDeviceGetAttribute' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
nvlink error   : Undefined reference to 'cudaDeviceSynchronize' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
nvlink error   : Undefined reference to 'cudaPeekAtLastError' in 'testing/CMakeFiles/thrust.test.cuda.fill.dir/cuda/fill.cu.o' (target: sm_53)
```

**Comment by allisonvacanti** [comment]
I've addressed this in my changes for #1159 by always disabling these arches by default. PR coming soon.

**Comment by brycelelbach** [comment]
These are all mobile GPU architectures. My guess is the `libdevice` in the main CUDA Toolkit does not have a variant for these architectures; there's probably some special `libdevice` that's part of the CUDA for Tegra toolkit. So, this isn't surprising.

**Comment by allisonvacanti** [comment]
Fixed in #1184.

There two sets of options that are incompatible now:

Set 1:
- `THRUST_ENABLE_COMPUTE_53`
- `THRUST_ENABLE_COMPUTE_62`
- `THRUST_ENABLE_COMPUTE_72`

Set 2:
- `THRUST_ENABLE_TESTS_WITH_RDC`
- `THRUST_ENABLE_EXAMPLES_WITH_RDC`

Turning on any option in Set 1 + any option in Set 2 is now a configuration error.

```
--- Enabled CUDA architectures: sm_35 sm_50 sm_52 sm_53 sm_60 sm_61 sm_62 sm_70 sm_72 sm_75 sm_80 compute_80
CMake Error at cmake/ThrustCUDAConfig.cmake:125 (message):
  THRUST_ENABLE_EXAMPLES_WITH_RDC is incompatible with
  THRUST_ENABLE_COMPUTE_53, since sm_53 does not support RDC.
Call Stack (most recent call first):
  CMakeLists.txt:85 (include)
```

Additionally, targets that require RDC (currently the tests in `testing/cuda`) will omit the flags for the 53, 62, or 72 architectures if they are enabled requested. A `NOTICE` CMake message is printed with this information when any of these archs are enabled.

```
--- Enabled CUDA architectures: sm_35 sm_50 sm_52 sm_53 sm_60 sm_61 sm_62 sm_70 sm_72 sm_75 sm_80 compute_80
sm_53 does not support RDC. Targets that require RDC will be built without support for this architecture.
sm_62 does not support RDC. Targets that require RDC will be built without support for this architecture.
sm_72 does not support RDC. Targets that require RDC will be built without support for this architecture.
```
